src/presenced.cpp

```diff
@@ -166,9 +166,44 @@ bool Client::setLastGreenVisible(bool enable)
 
 bool Client::requestLastGreen(Id userid)
 {
+    // Avoid send OP_LASTGREEN if user is ex-contact
+    if (isExContact(userid))
+    {
+        return false;
+    }
+
+    // Reset user last green or insert an entry in the map if not exists
+    std::map<uint64_t, time_t>::iterator it;
```

Comment:
All these lines can be simply replaced by `mPeersLastGreen[userid.val] = 0;`. Simplier and shorter :)